servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/MultiAddressUrlHttpClientTest.java

```diff
@@ -255,14 +255,33 @@ void requestWithAsteriskFormRequestTargetWithHostHeader() {
     void requestWithRedirect() throws Exception {
         StreamingHttpRequest request = client.get(format("http://%s/301", hostHeader))
                 .setHeader(X_REQUESTED_LOCATION, format("http://%s/200", hostHeader));  // Location for redirect
-        requestAndValidate(request, OK, "/200");
+        requestAndValidate(client, request, OK, "/200");
     }
 
     @Test
     void requestWithRelativeRedirect() throws Exception {
         StreamingHttpRequest request = client.get(format("http://%s/301", hostHeader))
                 .setHeader(X_REQUESTED_LOCATION, "/200");  // Location for redirect
-        requestAndValidate(request, OK, "/200");
+        requestAndValidate(client, request, OK, "/200");
+    }
+
+    @Test
+    void requestWithCustomDefaultHttpPort() throws Exception {
+        try (StreamingHttpClient client = HttpClients
+                .forMultiAddressUrl(ID).defaultHttpPort(serverPort).buildStreaming()) {
+            StreamingHttpRequest request = client.get(format("http://%s/200", serverHost));
+            requestAndValidate(client, request, OK, "/200");
+        }
+    }
+
+    @Test
+    void requestWithExplicitPortTakesPrecedenceOverCustomDefaultHttpPort() throws Exception {
+        int illegalPort = serverPort + 1;
```

Comment:
There is a risk that env where someone runs tests has an HTTP server on `serverPort + 1`. Consider using this trick to find an available port:

```java
    static int availablePort() throws IOException {
        ServerSocket s = new ServerSocket(0);
        int port = s.getLocalPort();
        s.close();
        return port;
    }
```